src/main/kotlin/net/atos/zac/mail/MailService.kt

```diff
@@ -0,0 +1,267 @@
+import jakarta.inject.Inject
+import net.atos.client.zgw.drc.DRCClientService
+import net.atos.client.zgw.drc.model.generated.EnkelvoudigInformatieObjectData
+import net.atos.client.zgw.shared.ZGWApiService
+import net.atos.client.zgw.shared.util.InformatieobjectenUtil
+import net.atos.client.zgw.zrc.model.Zaak
+import net.atos.client.zgw.ztc.ZTCClientService
+import net.atos.client.zgw.ztc.model.generated.InformatieObjectType
+import net.atos.zac.authentication.LoggedInUser
+import net.atos.zac.configuratie.ConfiguratieService
+import net.atos.zac.mail.model.Attachment
+import net.atos.zac.mail.model.Bronnen
+import net.atos.zac.mail.model.EMail
+import net.atos.zac.mail.model.EMails
+import net.atos.zac.mail.model.MailAdres
+import net.atos.zac.mailtemplates.MailTemplateHelper
+import net.atos.zac.mailtemplates.model.MailGegevens
+import net.atos.zac.util.JsonbUtil
+import nl.lifely.zac.util.AllOpen
+import nl.lifely.zac.util.NoArgConstructor
+import org.apache.commons.lang3.StringUtils
+import org.eclipse.microprofile.config.ConfigProvider
+import org.htmlcleaner.HtmlCleaner
+import org.htmlcleaner.PrettyXmlSerializer
+import java.io.ByteArrayOutputStream
+import java.io.IOException
+import java.time.LocalDate
+import java.util.Base64
+import java.util.UUID
+import java.util.function.Consumer
+import java.util.logging.Level
+import java.util.logging.Logger
+
+@ApplicationScoped
+@NoArgConstructor
+@AllOpen
+class MailService
+@Inject
+constructor(
+    private var configuratieService: ConfiguratieService,
+    private var zgwApiService: ZGWApiService,
+    private var ztcClientService: ZTCClientService,
+    private var drcClientService: DRCClientService,
+    private var mailTemplateHelper: MailTemplateHelper,
+    private var loggedInUserInstance: Instance<LoggedInUser>
+) {
+
+    companion object {
+        private val LOG = Logger.getLogger(MailService::class.java.name)
+
+        private val MAILJET_API_KEY =
+            ConfigProvider.getConfig().getValue("mailjet.api.key", String::class.java)
+        private val MAILJET_API_SECRET_KEY =
+            ConfigProvider.getConfig().getValue("mailjet.api.secret.key", String::class.java)
+
+        private const val HTTP_REDIRECT_RANGE = 300
+
+        // http://www.faqs.org/rfcs/rfc2822.html
+        private const val SUBJECT_MAXWIDTH = 78
+
+        private const val FONT_SIZE = 16f
+
+        private const val MEDIA_TYPE_PDF = "application/pdf"
+
+        private const val MAIL_VERZENDER = "Afzender"
+        private const val MAIL_ONTVANGER = "Ontvanger"
+        private const val MAIL_BIJLAGE = "Bijlage"
+        private const val MAIL_ONDERWERP = "Onderwerp"
+        private const val MAIL_BERICHT = "Bericht"
+    }
+
+    private val mailjetClient = MailjetClientHelper.createMailjetClient(MAILJET_API_KEY, MAILJET_API_SECRET_KEY)
+
+    val gemeenteMailAdres
+        get() = MailAdres(configuratieService.readGemeenteMail(), configuratieService.readGemeenteNaam())
```

Comment:
Probably generally better to do this sort of initialization inside of the constructor but let's leave that for now. It would then also need to work with the default constructor of course.